File: Server/ServerProject/src/main/java/com/dani/Token.java
package com.dani;

public class Token {
    public Object getTokenType;
    private String lexeme;
    private int tokenType;
    private int column;
    private int line;

    public Token(String lexeme, int tokenType, int column, int line) {
        this.lexeme = lexeme;
        this.tokenType = tokenType;
        this.column = column;
        this.line = line;
    }

    public String getLexeme() {
        return lexeme;
    }

    public void setLexeme(String lexeme) {
        this.lexeme = lexeme;
    }

    public int getTokenType() {
        return tokenType;
    }

    @Override
    public String toString() {
        return "Token{" +
                "lexeme='" + lexeme + '\'' +
                ", tokenType=" + tokenType +
                ", column=" + column +
                ", line=" + line +
                '}' + "\n";
    }
}

File: AppMovil/app/src/main/java/com/dani/appmovil/objects/Token.kt
package com.dani.appmovil.objects

import android.util.Log

data  class Token(val lexeme: String?, val tokenType: Int, val column: Int, val line:Int ) {
    private val TAG: String = "Token.class"

    fun printInfo() {
        Log.println(Log.INFO, TAG, "value: ${this.lexeme}, type: ${this.tokenType}, line: ${this.line}, column: ${this.column}")
    }

    fun getInfo(): String {
        return " ${this.lexeme}"
    }

    override fun toString(): String {
        return "Token(lexeme=$lexeme, tokenType=$tokenType, column=$column, line=$line, TAG='$TAG')"
    }


}